kyo-core/shared/src/main/scala/kyo/Clock.scala

```diff
@@ -104,21 +118,163 @@ object Clock:
     def let[A, S](c: Clock)(f: => A < S)(using Frame): A < S =
         local.let(c)(f)
 
+    /** Gets the current Clock instance from the local context.
+      *
+      * This is the primary way to access the current Clock instance when you only need to use it directly.
+      *
+      * @return
+      *   The current Clock instance
+      */
+    def get(using Frame): Clock < Any =
+        use(identity)
+
+    /** Uses the current Clock instance from the local context to perform an operation.
+      *
+      * This is useful when you want to perform multiple operations with the same Clock instance without having to repeatedly access it.
+      *
+      * @param f
+      *   A function that takes a Clock and returns an effect
+      * @return
+      *   The result of applying the function to the current Clock
+      */
+    def use[A, S](f: Clock => A < S)(using Frame): A < S =
+        local.use(f)
+
+    /** Runs an effect with a time-shifted Clock where time appears to pass faster or slower. This is particularly useful for testing
+      * time-dependent behavior without waiting for real time to pass.
+      *
+      * @param factor
+      *   The time scaling factor. Values > 1 speed up time, values < 1 slow down time
+      * @param v
+      *   The effect to run with scaled time
+      * @return
+      *   The result of running the effect with scaled time
+      */
+    def withTimeShift[A, S](factor: Double)(v: => A < S)(using Frame): A < (IO & S) =
+        if factor == 1 then v
```

Comment:
- Do you foresee any issues with imprecise `1.000001`? in the case of speeding up + slowing down?
- If a user speeds up with timeShift: 2.0, they would then need to run: .5 to get back normal speed. From my understanding the new clock will include multiple factors. Is there a better way to Reset to an 'unshifted' clock?

Reply:
The time shifting is scoped to the `v` computation. As soon as it finishes, the custom shifted clock is cleared and time returns to normal. I don't think precision would be an issue unless users nest time shifts, which I imagine won't be common.